map.go

```diff
@@ -1499,24 +1527,59 @@ type MapIterator struct {
 	err               error
 }
 
-func newMapIterator(target *Map) *MapIterator {
-	return &MapIterator{
+// keylessMapIterator iterates a Map defined with only values.
+// (Queue and Stack)
+//
+// See Map.Iterate.
+type keylessMapIterator struct {
```

Comment:
I'm not 100% convinced that making MapIterator an interface is currently worth it. Only `next()` is fundamentally different, and the kv vs. keyless split feels somewhat arbitrary. If we were planning to add many more different implementations, it would make more sense, but currently this is not being discussed.

See my comment on keylessMapIterator.next() below.

Reply:
I think that was how the code was structured originally, until Dylan suggested using an interface (after discussing it with me). I guess it doesn't make much difference either way.